app/models/branding_group.rb

```diff
@@ -69,7 +86,8 @@ def set_name_as_id
   # remove branding association on delete
```

Comment:
This function is not needed -- Rails/Mongoid will automatically remove the study-side associations on destroy